superset-frontend/src/explore/components/controls/SelectControl.jsx

```diff
@@ -128,6 +139,32 @@ export default class SelectControl extends React.PureComponent {
     this.props.onChange(onChangeVal, []);
   }
 
+  // Beware: This is acting like an on-click instead of an on-change
+  // (firing every time user chooses vs firing only if a new option is chosen).
+  onChangeDeprecated(val) {
+    // will eventually call `exploreReducer`: SET_FIELD_VALUE
+    const { valueKey } = this.props;
+    let onChangeVal = val;
+
+    if (Array.isArray(val)) {
+      let values;
```

Comment:
Do you need this variable?  Could you not just assign to onChangeVal directly in each if block?